### ISSUE TITLE
`UnsupportedOperationException` after adding a custom scenario

### ISSUE THREAD

**Issue by de-shyt** [JetBrains/lincheck#402]
The exception occurs when a custom scenario is added: 

```
java.lang.UnsupportedOperationException: can't get field offset on a record class: private final java.lang.reflect.Parameter[] java.lang.reflect.Executable$ParameterData.parameters
	at jdk.unsupported/sun.misc.Unsafe.objectFieldOffset(Unsafe.java:655)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.readField(ObjectTraverser.kt:201)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:74)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:63)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:63)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:42)
	at org.jetbrains.kotlinx.lincheck.IdeaPluginKt.createObjectToNumberMapAsArray(IdeaPlugin.kt:147)
	at org.jetbrains.kotlinx.lincheck.IdeaPluginKt.visualize(IdeaPlugin.kt:129)
	at org.jetbrains.kotlinx.lincheck.IdeaPluginKt.beforeEvent(IdeaPlugin.kt:83)
	at org.jetbrains.kotlinx.lincheck.strategy.managed.ManagedStrategy.beforeEvent(ManagedStrategy.kt:1314)
	at sun.nio.ch.lincheck.Injections.beforeEvent(Injections.java:331)
	at com.deshyt.ChannelTestBase.send(ChannelTestBase.kt:20)
	at org.jetbrains.kotlinx.lincheck.runner.TestThreadExecution406.run(Unknown Source)
	at org.jetbrains.kotlinx.lincheck.runner.FixedActiveThreadsExecutor.testThreadRunnable$lambda$10(FixedActiveThreadsExecutor.kt:172)
	at java.base/java.lang.Thread.run(Thread.java:1583)
java.lang.UnsupportedOperationException: can't get field offset on a record class: private final boolean java.lang.reflect.Executable$ParameterData.isReal
	at jdk.unsupported/sun.misc.Unsafe.objectFieldOffset(Unsafe.java:655)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.readField(ObjectTraverser.kt:208)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:74)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:63)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:63)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:105)
	at org.jetbrains.kotlinx.lincheck.ObjectTraverserKt.enumerateObjects(ObjectTraverser.kt:42)
	at org.jetbrains.kotlinx.lincheck.IdeaPluginKt.createObjectToNumberMapAsArray(IdeaPlugin.kt:147)
	at org.jetbrains.kotlinx.lincheck.IdeaPluginKt.visualize(IdeaPlugin.kt:129)
	at org.jetbrains.kotlinx.lincheck.IdeaPluginKt.beforeEvent(IdeaPlugin.kt:83)
	at org.jetbrains.kotlinx.lincheck.strategy.managed.ManagedStrategy.beforeEvent(ManagedStrategy.kt:1314)
	at sun.nio.ch.lincheck.Injections.beforeEvent(Injections.java:331)
	at com.deshyt.ChannelTestBase.send(ChannelTestBase.kt:20)
	at org.jetbrains.kotlinx.lincheck.runner.TestThreadExecution406.run(Unknown Source)
	at org.jetbrains.kotlinx.lincheck.runner.FixedActiveThreadsExecutor.testThreadRunnable$lambda$10(FixedActiveThreadsExecutor.kt:172)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

The code is stored [here](https://github.com/de-shyt/kotlin-channels/tree/buffered-channel-expandBuffer-UnsupportedOperationException) and can be reproduced by running [`Buffered1ChannelTest`](https://github.com/de-shyt/kotlin-channels/blob/c60ae4e86a4c611ee24d135b55e85573061181a7/src/test/kotlin/com/deshyt/ChannelsLincheckTest.kt#L24).
I am using blocking operations, and for this purpose I created a `blockingActor` method in the Lincheck library. The corresponding PR is [here](https://github.com/JetBrains/lincheck/pull/414/commits/4c546ebd01eca5b261058d632b5c80841fde09e2).

Lincheck version: 2.34
Lincheck plugin version: 0.9
Intellij IDEA version: 2024.2.2

**Comment by eupp** [comment]
Related to #367

**Comment by eupp** [comment]
It looks like we have several places in the codebase which traverse the object graph: `ObjectTraverser.kt`, `FieldSearchHelper.kt`, etc. The bug was fixed in only one of the places under #367. 

We need to extract common login (object graph traversal) into utility function, to avoid such problems in the future.

**Comment by dmitrii-artuhov** [comment]
https://github.com/JetBrains/lincheck/pull/420 resolves the issue
